pkg/daemon/ceph/client/upgrade_test.go

```diff
@@ -297,3 +298,63 @@ func TestGetRetryConfig(t *testing.T) {
 		assert.Equalf(t, tc.expectedDelay, actualDelay, "[%s] failed to get correct delays between retries", tc.label)
 	}
 }
+
+func TestOSDUpdateShouldCheckOkToStop(t *testing.T) {
+	clusterInfo := &ClusterInfo{}
+	lsOutput := ""
+	treeOutput := ""
+	context := &clusterd.Context{
+		Executor: &exectest.MockExecutor{
+			MockExecuteCommandWithOutputFile: func(command string, outFileArg string, args ...string) (string, error) {
+				t.Logf("command: %s %v", command, args)
+				if command != "ceph" || args[0] != "osd" {
+					panic("not a 'ceph osd' call")
```

Comment:
I think it's better to have the tests panic if something happens outside of their expected parameters so that if things have to change in the future, we don't run the risk of obfuscating any errors.